samples/filterSnippets.js

```diff
@@ -0,0 +1,351 @@
+  async function filterApplyLabel() {
+    const filter = {
+      label: 'labelled',
+    };
+    readWithFilter(filter);
+  }
+  // [END bigtable_filters_modify_apply_label]
+
+  // [START bigtable_filters_composing_chain]
+  async function filterChain() {
+    const filter = [
+      {
+        column: {
+          cellLimit: 1,
+        },
+      },
+      {
+        family: 'cell_plan',
+      },
+    ];
+    readWithFilter(filter);
+  }
+  // [END bigtable_filters_composing_chain]
+  // [START bigtable_filters_composing_interleave]
+  async function filterInterleave() {
+    const filter = {
+      interleave: [
+        {
+          value: 'true',
+        },
+        {column: 'os_build'},
+      ],
+    };
+    readWithFilter(filter);
+  }
+  // [END bigtable_filters_composing_interleave]
+  // [START bigtable_filters_composing_condition]
+  async function filterCondition() {
+    const filter = {
+      condition: {
+        test: [
+          {column: 'data_plan_10gb'},
+          {
+            value: 'true',
+          },
+        ],
+        pass: {
+          label: 'passed-filter',
+        },
+        fail: {
+          label: 'filtered-out',
+        },
+      },
+    };
+    readWithFilter(filter);
+  }
+  // [END bigtable_filters_composing_condition]
+
+  // [START bigtable_filters_limit_row_sample]
+  // [START bigtable_filters_limit_row_regex]
+  // [START bigtable_filters_limit_cells_per_col]
+  // [START bigtable_filters_limit_cells_per_row]
+  // [START bigtable_filters_limit_cells_per_row_offset]
+  // [START bigtable_filters_limit_col_family_regex]
+  // [START bigtable_filters_limit_col_qualifier_regex]
+  // [START bigtable_filters_limit_col_range]
+  // [START bigtable_filters_limit_value_range]
+  // [START bigtable_filters_limit_value_regex]
+  // [START bigtable_filters_limit_timestamp_range]
+  // [START bigtable_filters_limit_block_all]
+  // [START bigtable_filters_limit_pass_all]
+  // [START bigtable_filters_modify_strip_value]
+  // [START bigtable_filters_modify_apply_label]
+  // [START bigtable_filters_composing_chain]
+  // [START bigtable_filters_composing_interleave]
+  // [START bigtable_filters_composing_condition]
+  async function readWithFilter(filter) {
+    await table
+      .createReadStream({
+        filter,
+      })
+      .on('error', err => {
+        // Handle the error.
+        console.log(err);
+      })
+      .on('data', row => printRow(row.id, row.data))
+      .on('end', function() {
+        // All rows retrieved.
+      });
+  }
+
+  function printRow(rowkey, rowData) {
+    console.log(`Reading data for ${rowkey}:`);
+
+    for (const columnFamily of Object.keys(rowData)) {
+      const columnFamilyData = rowData[columnFamily];
+      console.log(`Column Family ${columnFamily}`);
+
+      for (const columnQualifier of Object.keys(columnFamilyData)) {
+        const col = columnFamilyData[columnQualifier];
+
+        for (let i = 0; i < col.length; i++) {
```

Comment:
unless you need the index, you could also do a `for (const cell of col) { ...` to make it more consistent